c10/core/CPUCachingAllocator.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <algorithm>
+#include <deque>
+#include <memory>
+#include <mutex>
+
+#include <c10/core/CPUAllocator.h>
+#include <c10/util/Exception.h>
+#include <c10/util/SmallVector.h>
+#include <c10/util/flat_hash_map.h>
+
+namespace c10 {
+
+class C10_API CPUCachingAllocator {
```

Comment:
I'd suggest not naming this a "Caching" allocator, as it makes it sound like this behaves similar to CUDACachingAllocator (when actually there is very little algorithmic similarity at all). This would be less of a big deal if this was in a mobile specific directory, but since the file is in `c10/core` people are much more likely to get the wrong idea. I'm not sure what a good replacement name is; I'm reminded of "BlackBox" nomenclature from the Caffe2 days; SizeArenaAllocator is another pretty descriptive alternative.